support-services/src/main/scala/com/gu/support/redemption/GetCodeStatus.scala

```diff
@@ -13,6 +13,7 @@ object GetCodeStatus {
   sealed abstract class RedemptionInvalid(val clientCode: String)
   case object NoSuchCode extends RedemptionInvalid("NO_SUCH_CODE")
   case object CodeAlreadyUsed extends RedemptionInvalid("CODE_ALREADY_USED")
+  case object InvalidReaderType extends RedemptionInvalid("INVALID_READER_TYPE")
```

Comment:
@rupertbates 
this change is causing a match error
```
[warn] /Users/john_duffell/code/support-frontend/support-frontend/app/controllers/RedemptionController.scala:230:56: match may not be exhaustive.
[warn] It would fail on the following input: InvalidReaderType
[warn]       _ <- EitherT(getCodeStatus(codeToCheck)).leftMap {
[warn]                  
```
on this line
https://github.com/guardian/support-frontend/blob/69996239214edcc7a42f6c8fb8b14e3ada0725f2/support-frontend/app/controllers/RedemptionController.scala#L230